File: LsSocial-Backend/LsSocial-Backend/Models/PostModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace LsSocial_Backend.Models
{
    public class PostModel
    {
        [Key]
        public Guid ID { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string Description { get; set; }

        [Required]
        public DateTime CreatedAt { get; set; }

        [Required]
        public DateTime UpdatedAt { get; set; }

        [Required]
        public Guid UserID { get; set; }

        [Required]
        public UserModel User { get; set; }
    }
}

File: LsSocial-Frontend/src/app/Models/post.model.ts
import { UserModel } from "./user.model";

export class PostModel {
    //#region Members
    public ID: string | undefined;
    public Name: string | undefined;
    public Description: string | undefined;
    public CreatedAt: Date | undefined;
    public UpdatedAt: Date | undefined;
    public UserID: string | undefined;
    public User: UserModel | undefined;
    //#endregion
}